01_fetch/src/fetch_noaa_nerrs.py

```diff
@@ -0,0 +1,51 @@
+import pandas as pd
+import os 
+import numpy as np
+import urllib
+import requests, json
+
+def fetch_metadata(station_id, metadata_outfile):
+    '''fetch metadata for NOAA NERRS station'''
+    metadata_url = f'https://api.tidesandcurrents.noaa.gov/mdapi/prod/webapi/stations/{station_id}/.json'
+    response = requests.get(metadata_url)
+    text = response.text
+    data = json.loads(text)
+    nested_list = pd.json_normalize(data, record_path=['stations'])
+    df = nested_list.stack().reset_index()
+    metadata = df.iloc[1:,1:].rename(columns={df.columns[1]:"NOAA NERRS metadata:",
+                                              df.columns[2]:(station_id)}).replace(".self","", regex=True)
```

Comment:
maybe we could name this something more specific like 'data_property'...just being nitpicky about semantics here
```suggestion
    metadata = df.iloc[1:,1:].rename(columns={df.columns[1]:"data_property",
                                              df.columns[2]:(station_id)}).replace(".self","", regex=True)
```